### ISSUE TITLE
EWS :: The request failed. Read timed out

### ISSUE THREAD

**Issue by Daij-Djan** [OfficeDev/ews-java-api#625]
Hi,
I have a server app that keeps a ExchangeService instance open for a long time. After a while it only throws timeout exceptions anymore.

Do I have to catch the exception and make a new ExchangeService instance? shouldnt the service refresh the connection it uses itself?

**Comment by celloni** [comment]
👍  same here.
@Daij-Djan Did you solve this via catching and making a new ExchangeService instance?

**Comment by Daij-Djan** [comment]
yes.. as  a workaround thats what I did.... I wrapped it in a AutoreconnectingExchangeService class 🍡

**Comment by arenaswan** [comment]
@Daij-Djan Same here。
could you show me the code of AutoreconnectingExchangeService?
please... i also faced the same issue .

**Comment by samrajcse** [comment]
Did anyone receive the code for auto reconnection?

**Comment by Daij-Djan** [comment]
dont have it anymore. it is with my previews company.

but it was a tiny 10 line bean that wrapped the actuall service in a try catch block - sorry.

**Comment by samrajcse** [comment]
Fine No problem.Thanks.
Regards,
*SAMRAJ.K*
